site-builder/server/server.ts

```diff
@@ -0,0 +1,586 @@
+// @ts-check
+
+import { createReadStream, ReadStream } from "fs";
+import fs from "fs/promises";
+import path from "path";
+import { PassThrough } from "stream";
+
+import { S3Client } from "@aws-sdk/client-s3";
+import { Upload } from "@aws-sdk/lib-storage";
+import { serve } from "@hono/node-server";
+import { initClient } from "@ts-rest/core";
+import { fetchRequestHandler, tsr } from "@ts-rest/serverless/fetch";
+import archiver from "archiver";
+import { Hono } from "hono";
+import mime from "mime-types";
```

Comment:
i chose to use `hono` as it seems like what the cool kids are using, but it doesn't work thaaat nicely with `ts-rest`.

Reply:
requires some manual setup, that's it